example-app/src/main/java/com/adyen/checkout/example/repositories/AddressLookupRepository.kt

```diff
@@ -15,19 +15,74 @@ import com.adyen.checkout.example.data.mock.model.MockAddressLookupResponse
 import com.squareup.moshi.JsonAdapter
 import com.squareup.moshi.Moshi
 import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onEach
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 class AddressLookupRepository @Inject constructor(
-    private val assetManager: AssetManager
+    assetManager: AssetManager
 ) {
+    private val mockDataService = MockDataService(assetManager)
 
-    fun getAddressLookupOptions(): List<LookupAddress> {
-        val mockDataService = MockDataService(assetManager)
+    private val addressLookupQueryFlow = MutableStateFlow<String?>(null)
+
+    @OptIn(FlowPreview::class)
+    val addressLookupOptionsFlow: Flow<List<LookupAddress>> = addressLookupQueryFlow
+        .filterNotNull()
+        .debounce(ADDRESS_LOOKUP_QUERY_DEBOUNCE_DURATION)
+        .map { query ->
+            queryAddressLookupOptions(query)
+        }
+
+    private val _addressLookupCompletionFlow: MutableStateFlow<AddressLookupCompletionState?> = MutableStateFlow(null)
+    val addressLookupCompletionFlow: Flow<AddressLookupCompletionState> = _addressLookupCompletionFlow
+        .asStateFlow()
+        .onEach { delay(ADDRESS_LOOKUP_COMPLETION_DELAY) }
+        .filterNotNull()
+
+    fun onQuery(query: String) {
+        addressLookupQueryFlow.tryEmit(query)
+    }
+
+    fun onAddressLookupCompleted(lookupAddress: LookupAddress) {
+        if (lookupAddress.id == ADDRESS_LOOKUP_ERROR_ITEM_ID) {
+            _addressLookupCompletionFlow.tryEmit(AddressLookupCompletionState.Error())
+        } else {
+            _addressLookupCompletionFlow.tryEmit(
+                AddressLookupCompletionState.Address(
```

Comment:
Can you just return `lookupAddress` here instead of fetching it by id from the list?